stlab/concurrency/future.hpp

```diff
@@ -17,12 +17,12 @@
 #include <mutex>
 #include <vector>
 
-#include <stlab/concurrency/config.hpp>
 #include <stlab/concurrency/executor_base.hpp>
 #include <stlab/concurrency/optional.hpp>
 #include <stlab/concurrency/task.hpp>
 #include <stlab/concurrency/traits.hpp>
 #include <stlab/concurrency/tuple_algorithm.hpp>
+#include <stlab/config.hpp>
```

Comment:
This worries me a little - but perhaps it shouldn't. In my experience I've put config files for a library before including any other header, just in case a header depends on `config` but fails to include it. Any concerns here?

Reply:
If a header depends on config.hpp, but fails to include it, that is a bug in that header and it isn't this header's job to work around that bug.

The fact that 'config.hpp' is generated doesn't it make it special in this regard. I went through this thought process and concluded it should be included just like any other header.